cypress-io/cypress
# feat: Cypress Cloud data on Specs page and Runs page use local Git data if available

- Closes https://github.com/cypress-io/cypress/issues/26693

### Additional details


Before this change, the Specs page and the Runs page did not use local Git data to determine which runs to show.  

#### Specs page

For the Specs page, an algorithm was used within the Cloud query to find runs for the current Branch name and then fall back to the `default` branch in the Cloud before that. This did not always match the actual Git tree the developer was using.   The new logic will use the same polling introduced with the Debug page to pull the "latest" runs that match the Git tree and pass those to new Cloud GQL fields to determine spec statuses for those runs.  The code changes under `packages/app/src/specs` will reflect the update to the GQL fields used.  The updates to `packages/app/src/pages/Specs/Index.vue` include the introduction of the `useRelevantRuns` composable.

#### Runs page

For the Runs page, it previously only just pulled the latest 10 runs from the Cloud project. It would cache these runs and then use a cursor pattern to pull new runs that were more recent than the latest run.  Now, the runs page preserves that logic as a fallback and it has been encapsulated into the `useProjectRuns` composable.  Another composable called `useGitTreeRuns` was added that uses the `useRelevantRuns` composable to pull runs relevant to the local Git tree and use those runs to display.  This will always just show the latest 10 runs for the local Git tree.  This may be expanded in a future initiative.  

For either scenario,  `RunCard` now uses a subscription to update itself if it is currently Running.  

#### Additional changes on this branch

* Updated the E2E test for the debug page to remove flake 


### Steps to test


#### Testing with Git 

* Find a project that is using Git
* Verify that the Specs page shows the run dot information and flaky information related to the local Git tree
  * Check the tooltips for the Runs and the flaky badges as well
  * Create a new branch, break a test, and record a run
  * Verify the Run status dots update while the tests are being recorded
* Verify the Runs page shows runs related to the local Git tree
  * Verify that a RUNNING run updates correctly.

#### Testing without Git
* Use a project without Git to record runs 
  * Or what I did was copy the directory for a project that already had runs and remove the `.git` folder
* Verify the Specs page does not show Cloud information (this is existing behavior)
* Verify the Runs page shows the latest runs for the project

### How has the user experience changed?


### PR Tasks
